packages/vm/src/cache.rs

```diff
@@ -137,7 +137,7 @@ where
         // Re-compile module from wasm
         let wasm = self.load_wasm(checksum)?;
         self.stats.misses += 1;
-        let module = compile(&wasm, Some(options.memory_limit))?;
+        let module = compile_and_use(&wasm, options.memory_limit)?;
```

Comment:
On cache miss we store with memory limit.
But on line 91 `compile_only`.

Meaning, first compile has no limit, but if we wipe out disk cache, then next recompile will have a memory limit?

I'm probably missing something, but wondering why there are two paths. Seems dangerous to ever use it without a limit, unless you throw away the module after that (don't store it)

Reply:
The memory limit is never stored in the cache. It lives in the store via `Tunables`. `compile_and_use` is a shortcut for `compile_only` followed by serialization, deserialization, adding a `make_runtime_store`.

Reply:
>  compile_and_use is a shortcut for compile_only followed by serialization, deserialization, adding a make_runtime_store.

Thanks for explaining, I get how this lives in the store. Why is there the serialization/deserialization everywhere? To ensure there are no bugs there (test all paths heavily) or for a deeper reason?

Reply:
I did not express myself very well. We avoid the serde. Those two paths both give you a usable Module:

1. `compile_only` (input Wasm) -> serialize artifact -> deserialize artifact + `make_runtime_store` (input memory limit)
2. `compile_and_use` (input Wasm + memory limit)

Where 1. is used when Wasm is compiled and cached in "store code" and 2. is used on a cache miss. In 2. we can avoid the serde.